From 3404eaa351f701b25103e4d208829f09bc3a0cb8 Mon Sep 17 00:00:00 2001
From: "Radu Pirea (NXP OSS)" <radu-nicolae.pirea@oss.nxp.com>
Date: Fri, 23 Jul 2021 15:14:01 +0300
Subject: [PATCH 15/16] s32g274abluebox3: boot from nfs with ramdisk

Add CONFIG_BOARD_NFS_BOOT_INTERFACE in order to configure the NFS boot
interface from menuconfig.
Define CONFIG_BOARD_USE_RAMFS_IN_NFSBOOT for S32G Bluebox3.

Issue: ALB-7424

Upstream-Status: Pending

Signed-off-by: Radu Pirea (NXP OSS) <radu-nicolae.pirea@oss.nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/Kconfig | 5 +++++
 include/configs/s32-gen1.h     | 4 ++++
 include/configs/s32.h          | 2 +-
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/arch/arm/cpu/armv8/s32/Kconfig b/arch/arm/cpu/armv8/s32/Kconfig
index f7c8822677..2410bc5bf5 100644
--- a/arch/arm/cpu/armv8/s32/Kconfig
+++ b/arch/arm/cpu/armv8/s32/Kconfig
@@ -203,6 +203,11 @@ config S32_CMU
 	  Checks if the frequency of a monitored clock (monitored_clock)
 	  is within a programmable frequency range specified by the user.
 
+config BOARD_NFS_BOOT_INTERFACE
+	string "Board NFS Boot Interface"
+	default "br0" if TARGET_S32G274ABLUEBOX3
+	default "eth0"
+
 source "board/freescale/Kconfig"
 source "board/microsys/mpxs32v234/Kconfig"
 
diff --git a/include/configs/s32-gen1.h b/include/configs/s32-gen1.h
index 33ea91b343..b70ef6363c 100644
--- a/include/configs/s32-gen1.h
+++ b/include/configs/s32-gen1.h
@@ -92,6 +92,10 @@
 #define CONFIG_SYS_FSL_FLASH0_SIZE      0x20000000
 #define QSPI_BASE_ADDR		        0x40134000
 
+#if defined(CONFIG_TARGET_S32G274ABLUEBOX3)
+#define CONFIG_BOARD_USE_RAMFS_IN_NFSBOOT
+#endif
+
 /* we include this file here because it depends on the above definitions */
 #include <configs/s32.h>
 
diff --git a/include/configs/s32.h b/include/configs/s32.h
index 0385f70240..330d6ba5c4 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -390,7 +390,7 @@
 	"netmask=255.255.255.0\0" \
 	"nfsbootargs=setenv bootargs console=${console},${baudrate} " \
 		"root=/dev/nfs rw " \
-		"ip=${ipaddr}:${serverip}::${netmask}::eth0:off " \
+		"ip=${ipaddr}:${serverip}::${netmask}::" CONFIG_BOARD_NFS_BOOT_INTERFACE ":off " \
 		"nfsroot=${serverip}:/tftpboot/rfs,nolock,v3,tcp " \
 		"earlycon " PFENG_EXTRA_BOOT_ARGS "\0" \
 	"loadtftpimage=tftp ${loadaddr} ${image};\0" \
-- 
2.17.1

